src/implementations/twig/components/social-media-share/social-media-share.html.twig

```diff
@@ -67,21 +51,51 @@
   {% if _links is not empty %}
     <ul class="ecl-social-media-share__list">
     {% for link in _links %}
-    {% if link.label is defined and link.label is not empty %}
-      {% set _link_extra_classes = 'ecl-social-media-share__link' %}
-      {% if link.extra_classes is not empty %}
-        {% set _link_extra_classes = _link_extra_classes ~ ' ' ~ link.extra_classes %}
-      {% endif %}
+      {# Backward compatibility provided by default(link) #}
       <li class="ecl-social-media-share__item">
+        {% set _link_extra_classes = 'ecl-social-media-share__link' %}
+        {% if link.extra_classes is not empty %}
+          {% set _link_extra_classes = _link_extra_classes ~ ' ' ~ link.extra_classes %}
+        {% endif %}
+        {% set _icon_extra_classes = 'ecl-social-media-share__icon' %}
+        {% if link.icon is not empty and link.icon.extra_classes is not empty %}
+          {% set _icon_extra_classes = _icon_extra_classes ~ ' ' ~ link.icon.extra_classes %}
+        {% endif %}
         {% include '@ecl/link/link.html.twig' with link|merge({
-          link: link|default({})|merge({
+          link: link.link|default(link)|merge({
             type: 'standalone',
           }),
+          icon: link.icon|default({})|merge({
+            extra_classes: _icon_extra_classes,
+          }),
           extra_classes: _link_extra_classes,
         }) only %}
       </li>
-    {% endif %}
     {% endfor %}
+
+    {% if _popover is not empty %}
```

Comment:
I guess that it could be the case, indeed. But that would mean that the popover could be located on every link (not only the last one), based on the data passed. There could also be multiple popover. This is not really the expected behaviour.
In the end, this twig template will most likely never be used (that's still a webtools widget), do I think that we shouldn't worrk to much about its optimisation.